app/models/transaction.rb

```diff
@@ -1,4 +1,5 @@
 class Transaction < ApplicationRecord
   belongs_to :user
   belongs_to :book
+  validates_numericality_of :quantity, greater_than: 0
```

Comment:
also needs a validation on total number of books. if i send more than the required amount it should yield error.